FeatureDetect/src/BuildOpenCV.cmake

```diff
@@ -0,0 +1,134 @@
+
+    if (NOT EXISTS ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip)
+        message(STATUS "OpenCV not downloaded into ${OPENCV_DOWNLOAD_DIR}")
+
+        message(STATUS "Downloading OpenCV")
+        file(
+                DOWNLOAD
+                https://github.com/opencv/opencv/archive/refs/tags/${OPENCV_VERSION}.zip
+                ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip
+                SHOW_PROGRESS
+        )
+        message(STATUS "Downloading OpenCV - done")
+    endif ()
+
+    message(STATUS "Extracting OpenCV")
+    file(
+            ARCHIVE_EXTRACT
+            INPUT ${OPENCV_DOWNLOAD_DIR}/${OPENCV_VERSION}.zip
+            DESTINATION ${OPENCV_DOWNLOAD_DIR}
+    )
+    message(STATUS "Extracting OpenCV - done")
+endif ()
+
+# Set toolchain-related flags
+if (CMAKE_GENERATOR)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_GENERATOR=${CMAKE_GENERATOR})
+endif ()
+if (CMAKE_TOOLCHAIN_FILE)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+endif ()
+if (CMAKE_C_COMPILER)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER})
+endif ()
+if (CMAKE_CXX_COMPILER)
+    list(APPEND TOOLCHAIN_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER})
+endif ()
+
+# Configure OpenCV build
+set(
+        OPENCV_CMAKE_ARGS  # https://docs.opencv.org/4.5.5/db/d05/tutorial_config_reference.html
+        # General
+        -DBUILD_LIST=${OPENCV_MODULES}
+        # Bundled components
+        # -DWITH_GTK=ON
+        -DBUILD_TESTS=OFF
+        -DBUILD_PERF_TESTS=OFF
+        -DBUILD_ANDROID_PROJECTS=OFF
+        -DBUILD_ANDROID_EXAMPLES=OFF
+        -DBUILD_opencv_apps=OFF
+        -DBUILD_JAVA=OFF
+        -DBUILD_FAT_JAVA_LIB=OFF
+        -DBUILD_KOTLIN_EXTENSIONS=OFF
+        -DBUILD_OBJC=OFF
+        -DBUILD_opencv_python2=OFF
+        -DBUILD_opencv_python3=OFF
+        # Disabled dependencies
+        -DWITH_1394=OFF
+        -DWITH_AVFOUNDATION=OFF
+        -DWITH_CAP_IOS=OFF
+        -DWITH_VTK=OFF
+        -DWITH_GSTREAMER=OFF
+        -DWITH_GTK=OFF
+        -DWITH_WIN32UI=OFF
+        -DWITH_FFMPEG=OFF
+        -DWITH_V4L=OFF
+        -DWITH_DSHOW=OFF
+        -DWITH_MSMF=OFF
+        -DWITH_DIRECTX=OFF
+        -DWITH_VA=OFF
+        -DWITH_VA_INTEL=OFF
+        -DWITH_LAPACK=OFF                 # TODO: make sure it's not used
+        -DWITH_PROTOBUF=OFF               # TODO: make sure it's not used
+        -DWITH_QUIRC=OFF
+        -DWITH_ANDROID_MEDIANDK=OFF
+        -DWITH_ANDROID_NATIVE_CAMERA=OFF
+        # Enabled dependencies
+        -DWITH_OPENCL=ON
+        -DWITH_OPENVX=ON                  # TODO: install OpenVX and set OPENVX_ROOT
+        -DWITH_EIGEN=ON                   # TODO: install Eigen (and probably use EIGEN_INCLUDE_PATH)
+        -DWITH_IPP=ON
+        # Cross-compilation handling
+        -DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY
+        -DANDROID_ABI=${ANDROID_ABI}
+        -DANDROID_ARM_NEON=${ANDROID_ARM_NEON}
```

Comment:
These two will be replaced with `ANDROID_ARGS` when you update the script as I described above

Reply:
done